mono/metadata/profiler.c

```diff
@@ -314,13 +314,12 @@ mono_profiler_get_coverage_data (MonoProfilerHandle handle, MonoMethod *method,
 
 	coverage_unlock ();
 
-	ERROR_DECL (error);
-	MonoMethodHeader *header = mono_method_get_header_checked (method, error);
-	mono_error_assert_ok (error);
+	MonoMethodHeaderSummary header;
 
-	guint32 size;
+	g_assert (mono_method_get_header_summary (method, &header));
```

Comment:
It is usually bad practice to put a function call with side effect in an assertion.

Reply:
I have no particular opinion on the subject, I just did it because I've seen a lot of other code in the runtime do it.

That said, I think if we ever want to compile without assertions, for whatever reason, we could do something like:

```c
#define g_assert(stuff) (void) (stuff)
```

That way, side effects are preserved.